refactor (request_)collection specs to use collection_shared_examples

ndushay opened the pull request:
## Why was this change made?

-    Reduce duplication of test code
-    Ensure Collection and RequestCollection models stay in sync.

Part of #70  (akin to PR #75, #76 for (Request)File and (Request)AdminPolicy models)

## Was the documentation (README, wiki) updated?

na

jcoyne commented on spec/cocina/models/request_collection_spec.rb:
```diff
@@ -1,228 +1,26 @@
-      end
-    end
+  it 'does not have Checkable model methods' do
+    item = described_class.new(required_properties)
```
Do we need a test for something that is not present?  There are an infinite number of things this object does not do and they aren't important to test for.

ndushay replied:
I can take it out;  I put it in to distinguish from Collection, but will remove.

ndushay commented:
I have now removed `it 'does not have Checkable model methods' do` tests for RequestCollection, and added a commit to also remove the same for RequestFile and RequestAdminPolicy.

ndushay commented on spec/cocina/models/collection_shared_examples.rb:
```diff
@@ -0,0 +1,290 @@
+              }
+            ]
+          },
+          description: {
```
I removed sourceId attribute as it's not in the struct and I'm not sure we associate collection objects with a sourceId ?
